test/dummy/app/views/scripts/reports.html.slim

```diff
@@ -76,5 +81,36 @@ coffee:
                     alert res.message
         )
 
+    showSettingsModal = (scriptId) ->
+      constants = null
+      script = null
+      onDone = _.after 2, ->
+        modal = window.scripts.newSettingsModal(
+                script
+                constants
+                {}
+        )
+      $.get(
+              '/scripts/constants.json'
+        (res) =>
+          if res.status == 'success'
+            constants = res.constants
+            onDone()
+          else
+            alert res.message
+      )
+      $.get(
+              "/scripts/#{scriptId}.json"
+        (res) =>
+          if res.status == 'success'
+            script = res.script
+            script.script_fields = _.toArray script.script_fields
+            onDone()
+          else
+            alert res.message
+      )
```

Comment:
If you're going to expose a public function, it seems like _this_ implementation would be better (instead of making the caller pass the script object and constants themselves).

Reply:
Yeah good idea. Changes are made.